packages/stg-evm-v2/ts-src/utils/helpers.ts

```diff
@@ -21,3 +26,136 @@ export const appendDependencies =
     (deploy: DeployFunction): DeployFunction => (
         (deploy.dependencies = [...(deploy.dependencies ?? []), ...dependencies]), deploy
     )
+
+/**
+ * Helper function that replaces any $ placeholders in str with the provided address
+ * @param str The string with placeholders
+ * @param address The address to replace the placeholders with
+ * @returns str with placeholders replaced with address
+ */
+export const fillAddress = (str: string, address: string) => {
+    // Ensure the address is 40 characters long (without the 0x prefix)
+    if (address.slice(0, 2) === '0x') {
+        address = address.slice(2)
+    }
+    if (address.length !== 40) {
+        throw new Error(`Invalid library address length ${address}`)
+    }
+
+    // Find the placeholder in the bytecode
+    const placeholder = `$`
+    while (str.includes(placeholder)) {
+        str = str.replace(placeholder, address)
+    }
+
+    return str
+}
+
+/**
+ * Helper function that saves deployment information to the deployments folder
+ * @returns the saved deployment
+ */
+export const saveDeployment = async ({
+    hre,
+    deploymentName,
+    deploymentContract,
+    abi,
+    creationBytecode,
+    deployedBytecode,
+    libraries,
+    args = [],
+    metadata,
+}: {
+    hre: HardhatRuntimeEnvironment
+    deploymentName: string
+    deploymentContract: Contract
+    abi: any
```

Comment:
`unknown` is safer than `any` so if the code allows, use `unknown`

Reply:
I ended up importing the ABI type instead